### ISSUE TITLE
Invoice pdf generation not in proper language

### ISSUE THREAD

**Issue by ksz2013** [magento/magento2#4409]
### Summary (*)
We are using a store with 3 store views, each store view a different language. Everything is ok on frontend, proper languages are displayed.
The problem is on the backend when we are generating the invoice pdf, (also shipping) the generated pdf is in the language of the admin logged in user. It is not taking in consideration the language of the store view where the order was placed. (If i select the desired language from backend for the logged in admin user the invoice pdf is generated in the atlanguage, but from what i am aware it should be translated to the store view language)

### Preconditions (*)
1. Magento 2.2.x & 2.3.x 

### Steps to reproduce (*)

1. Place order(on ENG store view)
2. Change locale of admin for any language (Danish in my case)
3. Go to the order and invoice
4. Download invoice PDF 

### Expected result (*)
Generated PDF is in the language of the store view where the order was placed like in email
![image](https://user-images.githubusercontent.com/51680850/69718237-8534bc80-1116-11ea-8e34-64f7e8080f56.png)

### Actual result (*)
Generated PDF is in the language of the admin logged in user
![image](https://user-images.githubusercontent.com/51680850/69718153-4c94e300-1116-11ea-9e00-c464a4f98156.png)

----------------------------------------------------------------------------------------------
The emulation on the invoice pdf is set:

Magento\Sales\Model\Order\Pdf\Invoice

Lines :130, 131

```
$this->_localeResolver->emulate($invoice->getStoreId());
$this->_storeManager->setCurrentStore($invoice->getStoreId());
```

`echo $this->_localeResolver->getLocale();` <--  The proper locale is returned, but nothing gets translated:  Ex:  __('Invoice # ')  -> is still english Invoice instead of the proper language translation

-> On the frontend proper language translation is returned

It this a bug, or some different setup needs to be done for backend translations,

Thanks,
Szabi

**Comment by ksz2013** [comment]
Hi guys,

Any update on this?

Thanks

**Comment by ksz2013** [comment]
Same happens on the latest 2.0.5 version, they are not translated. (This functionality worked well on magento 1). 

Any feedback would be appreciated. 

Best regards,

**Comment by ksz2013** [comment]
Same happens on the latest 2.0.5 version, they are not translated. (This functionality worked well on magento 1). 
Any feedback would be appreciated.

Best regards

**Comment by ksz2013** [comment]
Meanwhile made a fix for it, pull request was submitted.

**Comment by NadiyaS** [comment]
Hi @ksz2013 ,
thanks for reporting this issue and making PR. Please see comments in your PR and only after your changes we can start to process it.

**Comment by palamar** [comment]
According to [contributor guide](http://devdocs.magento.com/guides/v2.1/contributor-guide/contributing.html), tickets without response for two weeks should be closed.
Please feel free to reopen if it's needed.

**Comment by dankocrama** [comment]
i just check on magento 2.1.2 and i have the same issue is there a update for this ticket ?

**Comment by hezide** [comment]
i am also having this issue in 2.1.2

**Comment by zsoerenm** [comment]
+1

**Comment by BlitzInternet** [comment]
still a problem in v2.1.3

Update: here's a workaround that requires nothing but an entry into DI:

```xml
<type name="\Magento\Framework\Locale\ResolverInterface">
	<plugin name="mage_localeresolver_reload" type="Magento\Framework\Translate\Locale\Resolver\Plugin" sortOrder="1" disabled="false"/>
</type>
```

**Comment by diybook** [comment]
Hi  @BlitzInternet, your workaround works, except for the invoice head like: Order #, Order Date: or Invoice #. Any ideas how to fix this? Should the lines be added at the bottom or at the top of the di.xml? Does this change anything? Best regards.

**Comment by kanduvisla** [comment]
This is because of a strange issue in the translation. In the PDF they don't use the string `Order #`, but `Order #(SPACE)` (there is an extra space at the end). The same goes for `Invoice # ` and `Shipment # `. For example:

```
"Order # ","Bestelling # "
"Invoice # ","Factuur # "
"Shipment # ","Verzending # "

**Comment by kanduvisla** [comment]
@BlitzInternet : are there any known side-effects of this workaround? It seems like something is manipulated on a global scale now to address a very specific issue.

**Comment by BlitzInternet** [comment]
@kanduvisla we haven't experienced any up to now (in 2.1.4)

**Comment by lucscu** [comment]
> https://github.com/magento/magento2/issues/4409#issuecomment-276654512

in which di.xml?

**Comment by deepak-netwin** [comment]
@BlitzInternet Can you please mention in which di.xml we need to add that code?

**Comment by DeepParekhDev** [comment]
You can add this code in following file.
**vendor/magento/module-sales/etc/di.xml**
```
<type name="\Magento\Framework\Locale\ResolverInterface">
	<plugin name="mage_localeresolver_reload" type="Magento\Framework\Translate\Locale\Resolver\Plugin" sortOrder="1" disabled="false"/>
</type>
```

**Comment by PiroozMB** [comment]
@BlitzInternet your solution doesn't work. Any update on this?

**Comment by vlauria** [comment]
Hi,
i have this issue in 2.2.6.

**Comment by Dharmeshvaja91** [comment]
> Hi,
> i have this issue in 2.2.6.

try this 

### <type name="\Magento\Framework\Locale\ResolverInterface">
	<plugin name="mage_localeresolver_reload" type="Magento\Framework\Translate\Locale\Resolver\Plugin" sortOrder="1" disabled="false"/>
</type>

**Comment by eugenbg** [comment]
haha magento still hasn't fixed this issue after 3 years :)

**Comment by quyendam2612** [comment]
I confirmed there is still issue with translation in Magento 2.2.9

**Comment by kunzi** [comment]
This is still an issue in 2.3.2. PDF's language is determined by the language of the admin user and not the locale order was made with.

**Comment by julian-van-drielen** [comment]
This is still an occurring issue. Ticket should be reopened.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29221` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by mautz-et-tong** [comment]
As #19241 is the more substantial issue, I recommend to close this ticket here instead of the newer one and concentrate on #19241

**Comment by engcom-Charlie** [comment]
Hello @mautz-et-tong 
Okay, i am closing this issue as duplicate of https://github.com/magento/magento2/issues/19241.

**Comment by tellows** [comment]
Currently working on a migration to Magento 2.3.5 and this translation bug is still happening. I am afraid that this and other bugs (wrong date on invoice) will never be solved. Being forced by PayPal to do a migration to an untested and rarely working shopsystem...
